### DISCUSSION TITLE
[Unstable] Cargo salvage missions spawning cargo in odd places

### DISCUSSION THREAD

**Discussion by HawkinsThe71st** [FakeFishGames/Barotrauma#13736] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

While checking out the new cargo salvage missions I've noticed most of the time the cargo crates would spawn in odd places such as the middle of a sub or somewhere far away on the level. Unfortunately I did not save the level seeds where these issues occured but the issue seems fairly consistent.

One instance where the cargo spawned under the level?
![image](https://github.com/FakeFishGames/Barotrauma/assets/115734437/cd2d8dba-5c20-49fa-95ff-bdc13fa0dfe6)

One instance where the crate spawned in the middle of the Barsuk wreck:
![image](https://github.com/FakeFishGames/Barotrauma/assets/115734437/d418e80d-0d11-4a1e-8e2b-a9802cde96cb)
Also the crate contained a traitor mode item, not sure if its an issue:
![image](https://github.com/FakeFishGames/Barotrauma/assets/115734437/d1f4e0d7-3e5f-41c5-ac18-5a37da1ba9ed)

One instance where the cargo spawned off to the right side of the level:
![image](https://github.com/FakeFishGames/Barotrauma/assets/115734437/fb225a49-6df6-4771-b5f2-7b0e945e6fdd)

Another instance of the cargo spawning in odd location:

https://github.com/FakeFishGames/Barotrauma/assets/115734437/b24ed584-0a7b-4c0b-8d9e-c44c500141fb

At first I thought it might be due to some wrecks not having crate shelves but I had this instance where the cargo spawned someplace else in the level despite the wreck having shelves:
![image](https://github.com/FakeFishGames/Barotrauma/assets/115734437/e19f2879-b28e-49f1-ae3f-2e8605576ffa)

Even the artifact variant didn't spawn in the wreck:
![image](https://github.com/FakeFishGames/Barotrauma/assets/115734437/dab47387-d286-481c-923f-477d941a4c63)

### Reproduction steps

1. Start a cargo salvage mission in a campaign
2. Go to cargo mission marker

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Single player

### Version

v1.4.0.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
Thank you for the report! [I created an issue ticket about this now](https://github.com/FakeFishGames/Barotrauma/issues/13753), we'll look into it!

**Comment by Regalis11** [comment]
The issue has been fixed now in our private repo, fix coming up in the next unstable build.
